### PR TITLE
core/routing: new condition ipin

### PR DESCRIPTION
**Pull request by ftarnell**
Add a new router condition, "ipin": check if an IPv4 address is inside a given network.

Sample usage: 

```
./uwsgi --route-if 'ipin:${REMOTE_ADDR};127.0.0.0/8 return:200' --route-run return:404 --http :8080
```

All requests from localhost will return 200, while other requests will return 404.

### CODE REVIEW

File: core/routing.c [hunk 1]
```diff
@@ -1692,6 +1692,48 @@ static int uwsgi_route_condition_startswith(struct wsgi_request *wsgi_req, struc
...
+	in_addr_t ip, net, mask;
+
+        char *semicolon = memchr(ur->subject_str, ';', ur->subject_str_len);
+        if (!semicolon) return 0;
+
+        struct uwsgi_buffer *ub = uwsgi_routing_translate(wsgi_req, ur, NULL, 0, ur->subject_str, semicolon - ur->subject_str);
+        if (!ub || ub->pos > IP4_LEN) return -1;
+
+        struct uwsgi_buffer *ub2 = uwsgi_routing_translate(wsgi_req, ur, NULL, 0, semicolon+1, ur->subject_str_len - ((semicolon+1) - ur->subject_str));
+        if (!ub2 || ub2->pos > IP4_LEN) {
```
Comment (unbit) [inline core/routing.c]: pay attention here (and before for 'ub') as the case for ub->pos do not clear the buffers

**Comment by unbit** [comment]
what about renaming it to "ipv4in" ?

**Comment by ftarnell** [comment]
Okay:

* ub leak should be fixed
* I renamed ipin to ipv4in
* Added ipv6in as well

**Comment by unbit** [comment]
@xrmx can be backported too

**Comment by xrmx** [comment]
@unbit next time please consider squashing before merging

**Comment by unbit** [comment]
@xrmx yeah, i suppose you will go with chery-picking. next time i'll do it

**Comment by xrmx** [comment]
Backported, @ftarnell cool stuff!

**Pull request merged**